src/core/libraries/VestingMath.sol

```diff
@@ -120,36 +122,48 @@ library VestingMath {
     {
         uint256 blockTimestamp = block.timestamp;
 
-        // If the cliff time is in the future, return zero.
+        // If the cliff time is in the future, return the start unlock amount.
         if (cliffTime > blockTimestamp) {
-            return 0;
+            return unlockAmounts.start;
         }
 
-        // In all other cases, calculate the amount streamed so far. Normalization to 18 decimals is not needed
-        // because there is no mix of amounts with different decimals.
         unchecked {
-            // Calculate how much time has passed since the stream started, and the stream's total duration.
-            UD60x18 elapsedTime = ud(blockTimestamp - startTime);
-            UD60x18 totalDuration = ud(endTime - startTime);
+            uint128 unlockAmountsSum = unlockAmounts.start + unlockAmounts.cliff;
+
+            //  If the sum of the unlock amounts is greater than or equal to the deposited amount, return the deposited
+            // amount. The ">=" operator is used as a safety measure in case of a bug, as the sum of the unlock amounts
+            // should never exceed the deposited amount.
+            if (unlockAmountsSum >= depositedAmount) {
```

Comment:
Why is this necessary since we already have a check down here?

```solidity
if (streamedAmount > depositedAmount) {
    return withdrawnAmount;
}
```
If `unlockAmountsSum > depositedAmount` implies `streamedAmount > depositedAmount` so in case of a bug (i.e. ), user wont be able to withdraw any amount than already withdrawn any longer.

Therefore, this is not necessary.

Reply:
`unlockAmountsSum` and `streamedAmount`  are not the same.

my idea with checking the `unlockAmountsSum` to deposit amount, is to return as soon as possible in case they equal to not perform latter calculations (when `block.timestamp` is between the streamable range)

Reply:
Good thought. Agree with your explanation.